File: CS/Grid_BottomNewItemRow/GridSplitContainerDescendant/SplitContainerViewInfoDescendant.cs
using DevExpress.XtraEditors;
using DevExpress.XtraEditors.Drawing;
using System;
using System.Collections.Generic;
using System.Drawing;
using System.Linq;
using System.Text;


namespace Grid_BottomNewItemRow
{
    public class SplitContainerViewInfoDescendant : SplitContainerViewInfo
    {
        public SplitContainerViewInfoDescendant(SplitContainerControl container) : base(container) { }

        protected override void UpdatePanelBounds()
        {
            base.UpdatePanelBounds();

            Rectangle rect = Panel1Info.Bounds;
            rect.Height += Splitter.Bounds.Height;
            Panel1Info.Bounds = rect;
            rect = Splitter.Bounds;
            rect.Height = 0;
            Splitter.Bounds = rect;
                       
        }


    }
}

File: VB/Grid_BottomNewItemRow/GridSplitContainerDescendant/SplitContainerViewInfoDescendant.vb
Imports Microsoft.VisualBasic
Imports DevExpress.XtraEditors
Imports DevExpress.XtraEditors.Drawing
Imports System
Imports System.Collections.Generic
Imports System.Drawing
Imports System.Linq
Imports System.Text


Namespace Grid_BottomNewItemRow
	Public Class SplitContainerViewInfoDescendant
		Inherits SplitContainerViewInfo
		Public Sub New(ByVal container As SplitContainerControl)
			MyBase.New(container)
		End Sub

		Protected Overrides Sub UpdatePanelBounds()
			MyBase.UpdatePanelBounds()

			Dim rect As Rectangle = Panel1Info.Bounds
			rect.Height += Splitter.Bounds.Height
			Panel1Info.Bounds = rect
			rect = Splitter.Bounds
			rect.Height = 0
			Splitter.Bounds = rect

		End Sub


	End Class
End Namespace
